src/state.py

```diff
@@ -40,6 +41,63 @@ class Protocol(str, enum.Enum):
     HTTPS = "https"
 
 
+def _validate_hostname_value(value: str) -> str:
```

Comment:
Have we checked if pydantic provides this at all?

Reply:
I do not think pydantic provides hostname validation.
There is a issue for it, but I do think it is implemented: https://github.com/pydantic/pydantic/issues/8295